sdk/core/azure-core/inc/azure/core/http/curl/curl connection_pool.hpp

```diff
@@ -26,83 +26,161 @@ namespace Azure { namespace Core { namespace Test {
-#endif
-  public:
+  struct CurlTransportOptions
+  {
     /**
-     * @brief Mutex for accessing connection pool for thread-safe reading and writing.
+     * @brief The string for the proxy is used without any parsing to libcurl handle.
+     *
+     * @remark No validation for the string is done by the Azure SDK. More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_PROXY.html.
+     *
+     * @remark The default value is an empty string (no proxy).
+     *
      */
-    static std::mutex ConnectionPoolMutex;
-
+    std::string Proxy;
     /**
-     * @brief Keeps an unique key for each host and creates a connection pool for each key.
+     * @brief The string for the certificate authenticator is set to libcurl handle directly.
+     *
+     * @remark The Azure SDK will not check if the path is valid or not.
      *
-     * @detail This way getting a connection for a specific host can be done in O(1) instead of
-     * looping a single connection list to find the first connection for the required host.
+     * @remark The default is the built-in system specific path. More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_CAINFO.html
      *
-     * @remark There might be multiple connections for each host.
      */
-    static std::map<std::string, std::list<std::unique_ptr<CurlNetworkConnection>>>
-        ConnectionPoolIndex;
-
+    std::string CAInfo;
     /**
-     * @brief Finds a connection to be re-used from the connection pool.
-     * @remark If there is not any available connection, a new connection is created.
+     * @brief All HTTP requests will keep the connection channel open to the service.
      *
-     * @param request HTTP request to get #CurlNetworkConnection for.
+     * @remark The channel might be closed by the server if the server response has an error code.
+     * A connection won't be re-used if it is abandoned in the middle of an operation.
+     * operation.
      *
-     * @return #CurlNetworkConnection to use.
+     * @remark This option is managed directly by the Azure SDK. No option is set for the curl
+     * handle. It is `true` by default.
      */
-    static std::unique_ptr<CurlNetworkConnection> GetCurlConnection(Request& request);
+    bool HttpKeepAlive = true;
+    /**
+     * @brief This option determines whether curl verifies the authenticity of the peer's
+     * certificate.
+     *
+     * @remark The default value is `true`. More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_SSL_VERIFYPEER.html
+     *
+     */
+    bool SSLVerifyPeer = true;
 
     /**
-     * @brief Moves a connection back to the pool to be re-used.
+     * @brief Define the SSL options for the libcurl handle.
+     *
+     * @remark See more info here: https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html.
+     * The default option is all options `false`. This is a private implementation on Details
+     * namespace.
```

Comment:
The comment about details namespace doesn't make sense. Remove.

```suggestion
     * The default option is all SSL settings set to `false`.
```

Reply:
I will handle the `moving to Details`  in another PR